modules/sugar/src/test/ts/atomic/DimensionParseTest.ts

```diff
@@ -0,0 +1,35 @@
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import fc from 'fast-check';
+import * as Dimension from 'ephox/sugar/api/view/Dimension';
+
+UnitTest.test('All valid floats are valid', () => {
+  fc.assert(fc.property(fc.oneof(
+    // small to medium floats
+    fc.float(),
+    // big floats
+    fc.tuple(fc.float(), fc.integer(-20, 20)).map(([ mantissa, exponent ]) => mantissa * Math.pow(10, exponent))
+  ), (num) => {
+    const parsed = Dimension.parse(num.toString(), [ 'empty' ]).getOrDie();
+    Assert.eq('Number is unchanged by stringifying and parsing', num, parsed.value);
+    Assert.eq('Unit is empty', '', parsed.unit);
+    return true;
+  }));
+});
+
+UnitTest.test('All valid integers are valid', () => {
+  fc.assert(fc.property(fc.integer(), (num) => {
+    const parsed = Dimension.parse(num.toString(), [ 'empty' ]).getOrDie();
+    Assert.eq('Number is unchanged by stringifying and parsing', num, parsed.value);
+    Assert.eq('Unit is empty', '', parsed.unit);
+    return true;
+  }));
+});
+
+UnitTest.test('Accepts known units', () => {
+  Assert.succeeds('Accepts % in relative', () => Dimension.parse('1%', [ 'relative' ]).is({ value: 1, unit: '%' }));
+  Assert.succeeds('Accepts px in fixed', () => Dimension.parse('20px', [ 'fixed' ]).is({ value: 20, unit: 'px' }));
+  Assert.succeeds('Does not accept % in fixed', () => Dimension.parse('1%', [ 'fixed' ]).isNone());
+  Assert.succeeds('Accepts px in fixed/relative', () => Dimension.parse('20px', [ 'fixed', 'relative' ]).is({ value: 20, unit: 'px' }));
+  Assert.succeeds('Accepts unitless in unitless', () => Dimension.parse('1.4', [ 'empty' ]).is({ value: 1.4, unit: '' }));
```

Comment:
Should we just fallback to px if it is unitless since that is what browsers do. Do we need to know that it was without a unit when we parse?

Reply:
For some measurements (such as line-height) unitless is it's own thing that is distinct from pixel heights and has different behaviours.

Reply:
😅 Good, I was just about to say that too